validations/webhook.test.ts

```diff
@@ -0,0 +1,336 @@
+}
+
+interface WebhookKycStatusEventRecord {
+  event_id: string
+  provider: string // the field in the body sent by the provider
+  address: string
+  kyc_schema: string
+  kyc_status: string
+  event_timestamp: string
+  timestamp: string
+  provider_id: string // the URL path param, valora defined
+}
+
+async function validateTransferWebhook(
+  params: { transferId: string; address: string; status: TransferStatus },
+  retries: number,
+): Promise<void> {
+  const { transferId, address, status } = params
+  if (retries < 0) {
+    throw new Error('Webhook event could not be found')
+  }
+  await new Promise((resolve) => setTimeout(resolve, 2000))
+  const client = axios.create({
+    baseURL: WEBHOOK_RECIPIENT_BASE_URL,
+    validateStatus: () => true,
+  })
+  const response = await client.get(
+    `${WEBHOOK_RECIPIENT_HISTORY_ENDPOINT}/${config.providerId}`,
+  )
+  if (response.status !== 200) {
+    console.debug(
+      `Error fetching webhook history. ${retries} retries left: ${JSON.stringify(
+        response.data,
+      )}`,
+    )
+    return validateTransferWebhook(params, retries - 1)
+  }
+  const coorespondingWebhook = response.data.transferHistory.find(
+    (transfer: WebhookTransferEventRecord) =>
+      transfer.address === address &&
+      transfer.status === status &&
+      transfer.transfer_id === transferId,
+  )
+  if (!coorespondingWebhook) {
+    console.debug(`Could not find matching webhook. ${retries} retries left`)
+    return validateTransferWebhook(params, retries - 1)
+  }
+}
+
+async function validateKycWebhook(
+  params: { address: string; kycStatus: KycStatus; kycSchema: KycSchema },
+  retries: number,
+): Promise<void> {
+  const { kycSchema, address, kycStatus } = params
+  if (retries < 0) {
+    throw new Error('Webhook event could not be found')
+  }
+  await new Promise((resolve) => setTimeout(resolve, 2000))
+  const client = axios.create({
+    baseURL: WEBHOOK_RECIPIENT_BASE_URL,
+    validateStatus: () => true,
+  })
+  const response = await client.get(
+    `${WEBHOOK_RECIPIENT_HISTORY_ENDPOINT}/${config.providerId}`,
+  )
+  if (response.status !== 200) {
+    console.debug(
+      `Error fetching webhook history. ${retries} retries left: ${JSON.stringify(
+        response.data,
+      )}`,
+    )
+    return validateKycWebhook(params, retries - 1)
+  }
+  const coorespondingWebhook = response.data.kycHistory.find(
+    (transfer: WebhookKycStatusEventRecord) =>
+      transfer.address === address &&
+      transfer.kyc_status === kycStatus &&
+      transfer.kyc_schema === kycSchema,
+  )
+  if (!coorespondingWebhook) {
+    console.debug(`Could not find matching webhook. ${retries} retries left`)
+    return validateKycWebhook(params, retries - 1)
+  }
+}
+
+describe('webhooks', () => {
+  jest.setTimeout(15000)
+  const mockAccountData =
+    MOCK_FIAT_ACCOUNTS[
+      config.fiatAccountMock as keyof typeof MOCK_FIAT_ACCOUNTS
```

Comment:
not new to this PR but I feel like it'd be nicer if fiatAccountMock was already this type, so it didn't need to be cast each time it's used like this